modules/report/src/main/java/com/opengamma/strata/report/framework/format/AsciiTable.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.report.framework.format;
+
+import java.util.List;
+
+import com.google.common.base.Strings;
+
+/**
+ * An ASCII table generator.
+ * <p>
+ * Provides the ability to generate an ASCII table, typically used on the command line.
+ */
+final class AsciiTable {
+
+  //-------------------------------------------------------------------------
+  /**
+   * Restricted constructor.
+   */
+  private AsciiTable() {
+  }
+
+  /**
+   * Generates the ASCII table.
+   * 
+   * @param alignments  the table alignments
+   * @param headers  the table headers
+   * @param cells  the table cells
+   * @return the table
+   */
+  static String generate(
+      List<AsciiTableAlignment> alignments,
+      List<String> headers,
+      List<? extends List<String>> cells) {
+
+    int colCount = alignments.size();
+
+    // find max length of each column
+    int[] colLengths = new int[colCount];
+    for (int colIdx = 0; colIdx < colCount; colIdx++) {
+      colLengths[colIdx] = headers.get(colIdx).length();
+    }
+    for (List<String> row : cells) {
+      for (int colIdx = 0; colIdx < colCount; colIdx++) {
+        colLengths[colIdx] = Math.max(colLengths[colIdx], row.get(colIdx).length());
+      }
+    }
+
+    // write table
+    StringBuilder buf = new StringBuilder(2048);
```

Comment:
Wondering if we could be a bit more intelligent about the capacity here given that we've just worked out the width of each column, and we know the number of rows.